### PR TITLE
migrate pallet-nomination-pool-benchmarking to benchmarking syntax v2

### PR DESCRIPTION
**Pull request by clangenb**
Migrates pallet-nomination-pool-benchmarking to benchmarking syntax v2.

Part of:
* #6202

### CODE REVIEW

File: prdoc/pr_6302.prdoc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+title: migrate pallet-nomination-pool-benchmarking to benchmarking syntax v2
+doc:
+- audience: Runtime Dev
+  description: |-
+    Part of:
+    * #6202
```
Comment (gui1117) [inline prdoc/pr_6302.prdoc:6]:
```suggestion
    migrate pallet-nomination-pool-benchmarking to benchmarking syntax v2
```

File: substrate/frame/nomination-pools/benchmarking/src/inner.rs [hunk 2]
```diff
@@ -291,135 +293,142 @@ frame_benchmarking::benchmarks! {
...
 	}
 
-	bond_extra_other {
+	#[benchmark]
+	fn bond_extra_other() {
 		let claimer: T::AccountId = account("claimer", USER_SEED + 4, 0);
 
 		let origin_weight = Pools::<T>::depositor_min_bond() * 2u32.into();
-		let scenario = ListScenario::<T>::new(origin_weight, true)?;
+		let scenario = ListScenario::<T>::new(origin_weight, true).unwrap();
```
Comment (gui1117) [inline substrate/frame/nomination-pools/benchmarking/src/inner.rs:331]: same

File: substrate/frame/nomination-pools/benchmarking/src/inner.rs [hunk 3]
```diff
@@ -291,135 +293,142 @@ frame_benchmarking::benchmarks! {
...
 
-
-	unbond {
+	#[benchmark]
+	fn unbond() {
 		// The weight the nominator will start at. The value used here is expected to be
 		// significantly higher than the first position in a list (e.g. the first bag threshold).
 		let origin_weight = Pools::<T>::depositor_min_bond() * 200u32.into();
-		let scenario = ListScenario::<T>::new(origin_weight, false)?;
+		let scenario = ListScenario::<T>::new(origin_weight, false).unwrap();
```
Comment (gui1117) [inline substrate/frame/nomination-pools/benchmarking/src/inner.rs:400]: same

**Review by gui1117 — APPROVED** [review]
Some recommendation, but looks good anyway.

File: substrate/frame/nomination-pools/benchmarking/src/inner.rs [hunk 4]
```diff
@@ -270,19 +270,21 @@ impl<T: Config> ListScenario<T> {
...
+mod benchmarks {
+	use super::*;
+
+	#[benchmark]
+	fn join() {
 		let origin_weight = Pools::<T>::depositor_min_bond() * 2u32.into();
 
 		// setup the worst case list scenario.
-		let scenario = ListScenario::<T>::new(origin_weight, true)?;
+		let scenario = ListScenario::<T>::new(origin_weight, true).unwrap();
```
[the remark this answers is not in the document]
Comment (clangenb) [reply substrate/frame/nomination-pools/benchmarking/src/inner.rs:287]: Thanks for the review!

Yes, I know what you mean, but then I need to add an `Ok(())` at the end of every function, so @ggwpez advised me the opposite lately. XD

I had a hard time coming up with a meaningful strategy here, but I came up with the following:
1. If there is only one `unwrap`, I decide to unwrap, as this is the least verbose.
2. If there are multiple, I add the result type, as this is the least verbose.

I know this is inconsistent, but there seems to be no real consensus about this, so I think consistency is not possible anyway. 😅 

What do you think about this?

[the remark this answers is not in the document]
Comment (gui1117) [reply substrate/frame/nomination-pools/benchmarking/src/inner.rs:287]: I think PR is good then, no need to change it, unwrapping is ok, AFAIR client handles it ok.

**Review by re-gius — APPROVED** [review]

**Pull request merged**